Refactoring/improvements of AuthenticationManager

atomcat1978 opened the pull request:
Minor technical cleanup

Deprecated expireUserSessionCookie version with unused parameters.

Closes #43510

stianst reviewed (commented):
Added some comments inline; can you create a linked GitHub Issue and also do the sign-off of the commit?

atomcat1978 commented on services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java:
```diff
@@ -189,7 +190,7 @@ public static boolean isSessionValid(RealmModel realm, UserSessionModel userSess
         if (userSession.getNote(Details.IDENTITY_PROVIDER) != null) {
             String brokerAlias = userSession.getNote(Details.IDENTITY_PROVIDER);
-            if (realm.getIdentityProviderByAlias(brokerAlias) == null) {
+            if (realm.getFlowByAlias(brokerAlias) == null) {
```
Indeed, it should be getByAlias, my bad. `getIdentityProviderByAlias` is deprecated and `getByAlias` should be used.

```
    /**
     * @deprecated Use {@link IdentityProviderStorageProvider#getByAlias(String)} instead.
     */
    @Deprecated
    IdentityProviderModel getIdentityProviderByAlias(String alias);

```

atomcat1978 commented on services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java:
```diff
@@ -125,6 +123,9 @@
 import static org.keycloak.protocol.oidc.grants.device.DeviceGrantType.isOAuth2DeviceVerificationFlow;
+import static org.keycloak.sessions.CommonClientSessionModel.Action.*;
+import static org.keycloak.sessions.CommonClientSessionModel.Action.LOGGED_OUT;
+import static org.keycloak.sessions.CommonClientSessionModel.Action.LOGGING_OUT;
```
Agree, here there are especially a lot of enums defined with the same name from different enum types. Fixed the imports.

atomcat1978 commented on services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java:
```diff
@@ -223,7 +224,15 @@ public static boolean isClientSessionValid(RealmModel realm, ClientModel client,
+    /**
+     * @deprecated Use {@link #expireUserSessionCookie(KeycloakSession session, UserSessionModel userSession, RealmModel realm, UriInfo uriInfo)} instead.
+     */
     public static boolean expireUserSessionCookie(KeycloakSession session, UserSessionModel userSession, RealmModel realm, UriInfo uriInfo, HttpHeaders headers, ClientConnection connection) {
```
Done.

atomcat1978 committed:
Minor technical cleanup

Also added @Deprecated to expireUserSessionCookie version with unused parameters.

Closes #43510

Signed-off-by: Tamas Adam <tom@lambdac.no>

atomcat1978 commented:
> Can you squash the commits and add the DCO signature (https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#developers-certificate-of-origin)?

Done.
